extensions/sql-migration/src/dialog/tdeConfiguration/tdeConfigurationDialog.ts

```diff
@@ -245,15 +270,175 @@ export class TdeConfigurationDialog {
+
+		const validationRequiredLabel = _view.modelBuilder.text()
+			.withProps({
+				value: constants.TDE_VALIDATION_REQUIREMENTS_MESSAGE,
+				CSSStyles: {
+					...styles.BODY_CSS,
+					'margin': '4px 2px 4px 2px'
+				}
+			}).component();
+
+		const runValidationButton = _view.modelBuilder.button()
+			.withProps(
+				{
+					label: constants.TDE_VALIDATION_STATUS_RUN_VALIDATION,
+					enabled: true
+				}).component();
+
+		this._disposables.push(
+			runValidationButton.onDidClick(async (e) => {
+				let data = this._validationTable.data.map((e) => {
+					return [
+						e[0],
+						{
+							'icon': IconPathHelper.inProgressMigration,
+							'title': constants.TDE_VALIDATION_STATUS_RUNNING
+						},
+						constants.TDE_VALIDATION_STATUS_RUNNING
+					]
+				});
+				await this._validationTable.updateProperty('data', data);
+
+				let validationData = await this.migrationStateModel.runTdeValidation(
+					this.migrationStateModel.tdeMigrationConfig.getPendingNetworkPath());
+
+				let allValidationsSucceeded = true;
+
+				this._validationSuccessDescriptionErrorAndTips = validationData.result.map(e => {
+					return [
+						e.validationStatus.toString(),
+						e.validationDescription,
+						e.validationErrorMessage,
+						e.validationTroubleshootingTips
+					]
+				});
+
+				let res = validationData.result.map(e => {
+					if (e.validationStatus < 0) {
+						allValidationsSucceeded = false;
+					}
+
+					return [
+						e.validationTitle,
+						{
+							'icon': e.validationStatus > 0 ? IconPathHelper.completedMigration : IconPathHelper.error,
+							'title': e.validationStatusString
+						},
+						e.validationStatusString
+					]
+				});
+
+				await this._validationTable.updateProperty('data', res);
+
+				if (allValidationsSucceeded) {
+					this.migrationStateModel.tdeMigrationConfig.setLastValidatedNetworkPath(
+						this.migrationStateModel.tdeMigrationConfig.getPendingNetworkPath());
+					await this.updateUI();
+				}
+			}));
+
+		this._validationTable = this._createValidationTable(_view);
+
+		this._disposables.push(
+			this._validationTable.onRowSelected(
+				async (e) => {
+					const selectedRows: number[] = this._validationTable.selectedRows ?? [];
+
+					let message: string = '';
+					selectedRows.forEach((rowIndex) => {
+
+						let successful = this._validationSuccessDescriptionErrorAndTips[rowIndex][0] === "1" // Value will be "1" if successful
+						let description = this._validationSuccessDescriptionErrorAndTips[rowIndex][1];
+						let errorMessage = this._validationSuccessDescriptionErrorAndTips[rowIndex][2];
+						let tips = this._validationSuccessDescriptionErrorAndTips[rowIndex][3];
+
+						message = `Description:${EOL}${description}`;
+						if (!successful) {
+							message += `${EOL}${EOL}`;
+							if (errorMessage?.length > 0) {
+								message += `Error:${EOL}${errorMessage}${EOL}${EOL}`;
+							}
+
+							message += `Troubleshooting Tips:${EOL}${tips}`;
```

Comment:
are the strings "Description:", "Error:", "Troubleshooting Tips:" all user strings?  if so they would need to be added as resource strings for localization

Reply:
Good call, I forgot about these.